app/src/main/java/com/hamdan/forzenbook/domain/usecase/login/CreateAccountRequestUseCaseImpl.kt

```diff
@@ -1,17 +1,26 @@
 package com.hamdan.forzenbook.domain.usecase.login
 
+import com.hamdan.forzenbook.data.repository.AccountException
 import com.hamdan.forzenbook.data.repository.CreateAccountRepository
 
 class CreateAccountRequestUseCaseImpl(
     val repository: CreateAccountRepository
 ) : CreateAccountRequestUseCase {
-        override suspend fun invoke(
-            firstName: String,
-            lastName: String,
-            birthDay: String,
-            email: String,
-            location: String
-        ): Int {
-            return repository.createUser(firstName, lastName, birthDay, email, location)
+    override suspend fun invoke(
+        firstName: String,
+        lastName: String,
+        birthDay: String,
+        email: String,
+        location: String
+    ): CreateAccountResult {
+        return try {
+            if (repository.createUser(firstName, lastName, birthDay, email, location)) {
```

Comment:
When you remove the boolean this gets simpler, no reason for if here-- it'll just throw

Reply:
updated